workflows/prognostic_c48_run/runtime/segmented_run/logs.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python
+"""
+https://cloud.google.com/stackdriver/docs/solutions/gke/managing-logs
+
+Description of special payload strings
+https://cloud.google.com/logging/docs/structured-logging#special-payload-fields
+"""
+from dataclasses import dataclass
+import logging
+import re
+from enum import Enum
+from typing import Mapping, Optional
+import json
+from typing import Any, Iterable
+
+__all__ = ["handle_fv3_log"]
+
+
+class LineType(Enum):
+    # need to be letters for named groups to work
+    PYTHON_LOG = "python"
+    MAX_MIN = "max_min"
+    FV3_LOG = "fortran"
+
+
+# regexp for floating point number
+floating_point = r"[-+]?[0-9]*(\.[0-9]+(E-?\d+)?)?"
+
+REGEX = {
+    LineType.PYTHON_LOG: re.compile(
+        r"(?P<severity>(INFO|WARNING|ERROR|DEBUG|CRITICAL))"
+        ":"
+        r"(?P<module>.*?)"
+        ":"
+        r"(?P<message>.*)$"
+    ),
+    LineType.MAX_MIN: re.compile(
+        r"(?P<max_min_name>.*?)"
+        r"\s*max\s*=\s*"
+        + (r"(?P<max>" + floating_point + ")")
+        + r"\s*min\s*=\s*"
+        + (r"(?P<min>" + floating_point + ")")
+    ),
+    LineType.FV3_LOG: re.compile("(?P<message>.*$)"),
+}
+
+
+@dataclass
+class LogLine:
+    type: LineType
+    data: dict
+    line: str
+
+
+def parse_line(line):
+    for line_type in LineType:
+        match = REGEX[line_type].match(line)
+        if match:
+            return LogLine(line_type, match.groupdict(), line)
+    raise ValueError(f"Unable to parse line: {line}")
+
+
+class Handler:
+    """Handler handels the parsed lines produces by the log stream
```

Comment:
```suggestion
    """Handler handles the parsed lines produces by the log stream
```

Reply:
I actually meant this guy:

![image](https://user-images.githubusercontent.com/1386642/162484523-4e436bf9-b58d-40e1-bf82-cdd8cb9c2de0.png)